API simplification

flo-dup opened the pull request:
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**Does this PR already have an issue describing the problem ?**
Yes, fixes #46 and fixes #151

geofjamg commented:
About SingleLineDiagram utility class:
 - Use `SmartVoltageLevelLayoutFactory` by defautl, otherwise, in case of missing extensions, we have an error
 - Use `TopologicalStyleProvider` by default
 - Embed the CSS by default to have a standalone SVG
 - adaptCellHeightToContent to true by default (to change in `LayoutParameter`)
 -  Duplicate all methods with `Path` to also have `Writer`
 - Add an ultimate simplistic method `SingleLineDiagram.draw(Network network, String id, String svgFile)`

flo-dup commented:
> About SingleLineDiagram utility class:
>     * Use `SmartVoltageLevelLayoutFactory` by defautl, otherwise, in case of missing extensions, we have an error
>     * Use `TopologicalStyleProvider` by default
>     * Embed the CSS by default to have a standalone SVG
>     * adaptCellHeightToContent to true by default (to change in `LayoutParameter`)
>     * Duplicate all methods with `Path` to also have `Writer`
>     * Add an ultimate simplistic method `SingleLineDiagram.draw(Network network, String id, String svgFile)`

I did these changes, and also:
* flipped ComponentLibrary / LayoutParameters arguments in draw methods, as the latter is more often changed as the former I think
* I removed useName parameter, which is in fact a rendering parameter and should not appear directly in the API; it is now in LayoutParameters, as numerous other rendering parameters

Note that I added methods with two `Writer` arguments (not just one) instead of `Path`: one is needed for the json metadata, in addition to the one for the svg itself.
